src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -39,12 +40,17 @@ internal Metric AddMetricWithNoViews(Instrument instrument)
             var meterName = instrument.Meter.Name;
             var metricName = instrument.Name;
             var metricStreamName = $"{meterName}.{metricName}";
+            var instrumentIdentity = new InstrumentIdentity(meterName, metricName, instrument.Unit, instrument.Description, instrument.GetType());
             lock (this.instrumentCreationLock)
             {
+                if (this.instrumentIdentityToMetric.TryGetValue(instrumentIdentity, out var existingMetric))
+                {
+                    return existingMetric;
+                }
+
                 if (this.metricStreamNames.Contains(metricStreamName))
                 {
-                    OpenTelemetrySdkEventSource.Log.MetricInstrumentIgnored(metricName, instrument.Meter.Name, "Metric name conflicting with existing name.", "Either change the name of the instrument or change name using View.");
-                    return null;
+                    OpenTelemetrySdkEventSource.Log.DuplicateMetricInstrument(metricName, meterName, "Metric instrument has the same name as an existing one but differs by description, unit, or instrument type.", "Either change the name of the instrument or use MeterProviderBuilder.AddView to resolve the conflict.");
```

Comment:
Should we update the message? The user doesn't really need to resolve any conflict here, right?

Reply:
I think we can also mention in the log that "SDK will still exporter duplicates". Else it is not clear if SDK is going to drop it or pass it to exporters.